src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt

```diff
@@ -124,6 +136,63 @@ private suspend fun writeOutputVCF(
 
 }
 
+/**
+ * Writes the merged VCF files. This is used when a bedfile is provided.
+ * The bedfile is used to create one VCF file per range in the bedfile.
+ */
+private suspend fun writeOutputVCF(
+    outputFile: String,
+    bedfile: String,
+    samples: List<String>,
+    variantContextChannel: Channel<Deferred<List<VariantContext>>>
+) {
+
+    val ranges = readBedfile(bedfile)
+
+    val header = VCFHeader(createGenericVCFHeaders(samples))
+
+    var writePositionRange: PositionRange? = null
+    var writer: VariantContextWriter? = null
+    for (deferred in variantContextChannel) {
+
+        val variantContexts = deferred.await()
+        variantContexts.forEach { variantContext ->
+
+            val currentPositionRange = PositionRange(variantContext.contig, variantContext.start, variantContext.end)
+
+            // If the current position range doesn't overlap with the current write position range
+            // then close the current writer and open a new one
+            if (writePositionRange == null || !writePositionRange!!.overlapping(currentPositionRange)) {
+
+                // Close the previous writer
+                writer?.close()
+                writer = null
+
+                // Find the range in the bedfile that overlaps with the current position range
+                // and create a new writer for that range. If no range is found, then don't create
+                // a writer.
+                writePositionRange = ranges.find { it.overlapping(currentPositionRange) }
```

Comment:
writePositionRange could be multiple ranges?

Reply:
These are SNPs, so only one position per variant